temporian/implementation/numpy/data/event.py

```diff
@@ -0,0 +1,50 @@
+from typing import Dict, List, Tuple
+
+import numpy as np
+
+from temporian.core.data.event import Event
+from temporian.core.data.feature import Feature
+from temporian.implementation.numpy.data.sampling import NumpySampling
+
+
+class NumpyFeature:
+    def __init__(self, name: str, index: List[str], data: np.ndarray) -> None:
+        if len(data.shape) > 1:
+            raise ValueError(
+                "NumpyFeatures can only be created from flat arrays. Passed"
+                f" input's shape: {len(data.shape)}"
+            )
+
+        self.name = name
+        self.index = index
+        self.data = data
+        self.dtype = self.data.dtype
+
+    def schema(self) -> Feature:
+        return Feature(
+            name=self.name, dtype=self.dtype, sampling=self.index.names
```

Comment:
We have our own dtype. There need to be some conversion from/to the numpy dtype.

Reply:
Will address in upcoming PR